tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+- For each test, we're going to need to deploy our contracts and mint an NFT. To do this, we can take advantage of the [`beforeEach` hook provided by Mocha](https://mochajs.org/#hooks){target=_blank}
+- As we'll be minting an NFT for each test, we'll need to have a `tokenUri`. The `tokenUri` that we'll use for our examples will be for Daizy, our first Dizzy Dragon NFT. The `tokenUri` will be set to `"https://gateway.pinata.cloud/ipfs/QmTCib5LvSrb7sshLhLvzmV7wdSdmSt3yjB4dqQaA58Td9"`
+
+So, now that we have a game plan, let's implement it! In the `test_NftMarketplace.js` file, we can add the following code:
+
+```js
+const NftMarketplace = artifacts.require("NftMarketplace");
+const DizzyDragons = artifacts.require("DizzyDragons");
+
+contract("NftMarketplace", (accounts) => {
+  const tokenUri =
+    "https://gateway.pinata.cloud/ipfs/QmTCib5LvSrb7sshLhLvzmV7wdSdmSt3yjB4dqQaA58Td9";
+  let nftMarketplace;
+  let dizzyDragonsNft;
+  let mintedNft;
+
+  beforeEach(async () => {
+    // Deploy the marketplace
+    nftMarketplace = await NftMarketplace.deployed();
+    // Deploy a Dizzy Dragon NFT
+    dizzyDragonsNft = await DizzyDragons.deployed();
+    // Mint Daizy the Dizzy Dragon NFT
+    mintedNft = await dizzyDragonsNft.mint(tokenUri);
+  });
+
+  // TODO: Add tests here
+});
+```
+
+!!! note
+    You don't have to import Mocha or Web3 as both packages comes out-of-the-box with Truffle!
+
+Now we're ready to start writing our tests!
+
+### Test Minting NFTs {: #test-minting-nfts }
+
+For our first test, let's make sure that we are minting our new Dizzy Dragon NFT as expected. We'll use the event logs from the transaction to ensure that the `NftMinted` event of the `DizzyDragons` contract has been emitted. We'll also test that the token ID of the NFT is `1`, as we are deploying the `DizzyDragons` contract fresh before each test, so it should always be the first NFT minted.
+
+In place of the `// TODO: Add tests here` comment, you can add the following test:
+
+```js
+  it("should mint a new Dizzy Dragon NFT", async () => {
+      // Access the logs of the mint transaction
+      // Remember: mintedNft was created in the beforeEach function
+      const nftMintedLog = mintedNft.logs[2];
+
+      const event = nftMintedLog.event;
+      const tokenId = nftMintedLog.args[0].toNumber();
+
+      // Use Mocha's assert to test that the NftMinted event was emitted
+      // and the token ID of the NFT is 1
+      assert.equal(event, "NftMinted");
+      assert.equal(tokenId, 1);
+  });
+```
+
+Assuming your [Moonbeam development node is up and running](#start-development-node), you can run the test with the following command:
+
+```
+npx truffle test --network dev
```

Comment:
Why  not run it and show the result? If you are going to do this at the very end I would say so